contrib/platform/test/com/sun/jna/platform/linux/LibRTTest.java

```diff
@@ -0,0 +1,120 @@
+/* Copyright (c) 2020 Daniel Widdis, All Rights Reserved
+ *
+ * The contents of this file is dual-licensed under 2
+ * alternative Open Source/Free licenses: LGPL 2.1 or later and
+ * Apache License 2.0. (starting with JNA version 4.0.0).
+ *
+ * You can freely decide which license you want to apply to
+ * the project.
+ *
+ * You may obtain a copy of the LGPL License at:
+ *
+ * http://www.gnu.org/licenses/licenses.html
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "LGPL2.1".
+ *
+ * You may obtain a copy of the Apache License at:
+ *
+ * http://www.apache.org/licenses/
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "AL2.0".
+ */
+package com.sun.jna.platform.linux;
+
+import static com.sun.jna.platform.linux.ErrNo.EEXIST;
+import static com.sun.jna.platform.linux.Fcntl.O_CREAT;
+import static com.sun.jna.platform.linux.Fcntl.O_EXCL;
+import static com.sun.jna.platform.linux.Fcntl.O_RDWR;
+import static com.sun.jna.platform.linux.Fcntl.S_IRWXU;
+import static com.sun.jna.platform.linux.Mman.MAP_FAILED;
+import static com.sun.jna.platform.linux.Mman.MAP_SHARED;
+import static com.sun.jna.platform.linux.Mman.MS_SYNC;
+import static com.sun.jna.platform.linux.Mman.PROT_READ;
+import static com.sun.jna.platform.linux.Mman.PROT_WRITE;
+import static org.junit.Assert.assertNotEquals;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.junit.Test;
+
+import com.sun.jna.Native;
+import com.sun.jna.Pointer;
+import com.sun.jna.platform.unix.LibCAPI.size_t;
+import com.sun.jna.platform.unix.LibCUtil;
+
+import junit.framework.TestCase;
+
+public class LibRTTest extends TestCase {
+
+    public static LibC LIBC = LibC.INSTANCE;
+    public static LibRT LIBRT = LibRT.INSTANCE;
+
+    @Test
+    public void testMmapToShm() throws IOException {
+        // Get a suitably random filename of the form "/somename"
+        File f = File.createTempFile("mmapToShm", "test");
+        String share = "/" + f.getName();
```

Comment:
This is not a file, but a virtual object (at least that is my reading from the manpage). It is possible to map a file, but that is `open` with `mmap`, not `shm_open`.

Reply:
Correct.  I was just leveraging the `createTempFile()` code to generate a somewhat random filename (e.g., `mapToShm192837461234test`).  The file `f` would actually briefly exist just long enough to get its name, and then be deleted.  What I really need is just a name `"/something"` for the virtual file.   I'm dual-purposing this name as also the "text"/bytes to write/read from the share as part of the test, and I thought it would be more useful for that text to be somewhat randomly generated.

I could just use `Random` to generate some numbers (or dig into `createTempFile()` source code to see what it uses) but since this is just test code, I took the quick-and-dirty route.